Adds support for basic auth application

vijetm opened the pull request.

quantumew commented on examples/okta_app_basic_auth/README.md:
```diff
@@ -0,0 +1,6 @@
+# okta_app_bookmark
```
Can you update this title?

quantumew commented on examples/okta_app_basic_auth/basic_updated.tf:
```diff
@@ -0,0 +1,22 @@
+  email       = "test-acc-replace_with_uuid@example.com"
+}
+
+resource "okta_group" "group" {
```
Looks like this group is unused, can you pass it into the app or nix it?

quantumew commented on okta/provider.go:
```diff
@@ -160,6 +162,7 @@ func Provider() terraform.ResourceProvider {
 			"okta_saml_idp_signing_key":      deprecateIncorrectNaming(resourceIdpSigningKey(), idpSamlKey),
 			"okta_social_idp":                deprecateIncorrectNaming(resourceIdpSocial(), idpSocial),
 			"okta_bookmark_app":              deprecateIncorrectNaming(resourceAppBookmark(), appBookmark),
+			"okta_basic_auth_app":            deprecateIncorrectNaming(resourceAppBasicAuth(), appBasicAuth),
```
You can drop this line. Some resources were incorrectly named and thus we are still supported them with a deprecation message. All new resources can avoid this.

quantumew commented on okta/resource_okta_app_basic_auth.go:
```diff
@@ -0,0 +1,131 @@
+	return &schema.Resource{
+		CustomizeDiff: func(d *schema.ResourceDiff, v interface{}) error {
+			return nil
+		},
```
You can drop this function, looks like it is noop.

vijetmahabaleshwar-okta commented:
@quantumew - I have addressed your comments and updated the website docs. Not sure if these are the only files that needs to be updated. Let me know if you need any more changes.

vijetm commented:
@quantumew - I have addressed your comments and updated the website docs. Not sure if these are the only docs/files that needs to be updated. Let me know if you need any more changes.

quantumew commented on website/docs/r/app_basic_auth.html.markdown:
```diff
@@ -0,0 +1,51 @@
+
+* `url` - The URL of the sign-in page for basic auth app.
+
+* `url` - The URL of the authenticating site for basic auth app.
```
I think this should be `auth_url`. Awesome work on these docs!

quantumew commented:
<img width="1019" alt="Screen Shot 2019-11-12 at 8 51 54 AM" src="https://user-images.githubusercontent.com/14934291/68765819-4ade0200-05e3-11ea-9f93-9cb50d6ab482.png">

varunmukka-okta commented:
👏 @vijetm
